adam/core/urdf_tree.py

```diff
@@ -0,0 +1,140 @@
+# Copyright (C) 2021 Istituto Italiano di Tecnologia (IIT). All rights reserved.
+# This software may be modified and distributed under the terms of the
+# GNU Lesser General Public License v2.1 or any later version.
+
+import logging
+from dataclasses import dataclass, field
+from os import error
+
+from prettytable import PrettyTable
+from urdf_parser_py.urdf import URDF
+
+
+@dataclass
+class Tree:
+    joints: list = field(default_factory=list)
```

Comment:
If you:

```python
from typing import List
```

you can constrain:

```suggestion
    joints: List[str] = field(default_factory=list)
```